### PR TITLE
age_calculate, dob_from_chi and age_from_chi added

### PR DESCRIPTION
**Pull request by Tina815**

### CODE REVIEW

File: tests/testthat/test-age_calculate.R [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+test_that("Age is calculated correctly for leap years", {
+  expect_equal(age_calculate(as.Date("2020-02-29"), as.Date("2022-02-28"),
+                             units = "years", date_class = "period",
+                             round_down = TRUE), 1)
+  expect_equal(age_calculate(as.Date("2020-02-29"), as.Date("2022-03-01"),
+                             units = "years", date_class = "period",
+                             round_down = TRUE), 2)
+})
+
```
Comment (Moohan) [inline tests/testthat/test-age_calculate.R:33]: Probably prudent to have some additional tests here e.g. counts of days with a leap day in it - "2020-02-20":"2020-03-01" should be 10 days.

File: R/age_calculate.R [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  if (date_class == "period" & !units %in% c("week", "weeks")) {
+    age_interval <- lubridate::as.period(age_interval, unit = units)
+  }
+
+  if (round_down) {
+    age_interval %/% unit_time
+  } else {
+    age_interval / unit_time
+  }
```
Comment (Moohan) [inline R/age_calculate.R:75]: Explicitly return.
```suggestion
  age_interval <- dplyr::if_else(round_down,
      age_interval %/% unit_time,
    age_interval / unit_time)

  return(age_interval)
```

File: R/age_calculate.R [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+#'
+#' @param start A start date (typically date of birth) which must be supplied with \code{Date} or \code{POSIXct}
+#' class. \code{\link[base:as.Date]{as.Date()}},
+#' \code{\link[lubridate:ymd]{lubridate::dmy()}} and
+#' \code{\link[base:as.POSIXlt]{as.POSIXct()}} are examples of functions which
+#' can be used to store dates as an appropriate class.
+#' @param end An end date which must be supplied with \code{Date} or \code{POSIXct} class. Defaulted as Sys.time()
+#' @param units Type of units to be used. Seconds, minutes, hours, days, weeks, months and years are accepted.
+#' @param round_down Should returned ages be rounded down to the nearest whole number. Defaulted as `TRUE`.
+#' @param date_class Type of `lubridate` date class. Accepted arguments are "period" and "duration".
```
Comment (Moohan) [inline R/age_calculate.R:15]: I'm not sure I understand the purpose of `date_class`. I would expect this function to return a numeric. What is the default return type and how does `date_class` change that?

File: R/dob_from_chi.R [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  }
+
+  guess_dob <- dob_from_chi(
+    chi_number = chi_number,
+    min_date = min_date.age,
+    max_date = max_date.age,
+    chi_check = chi_check
+  )
+
+  guess_age <- age_calculate(guess_dob, ref_date)
```
Comment (Moohan) [inline R/dob_from_chi.R:197]: From my reading of `age_calculate` isn't the parameter `date_class` required? (see my other comment on `date_class`

File: tests/testthat/test-age_calculate.R [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "hours", date_class = "period",
+                             round_down = TRUE), 9240)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "minutes", date_class = "period",
+                             round_down = TRUE), 554400)
+  expect_equal(age_calculate(as.Date("2021-02-01"), as.Date("2022-02-21"),
+                             units = "seconds", date_class = "period",
+                             round_down = TRUE), 33264000)
+})
```
Comment (Moohan) [inline tests/testthat/test-age_calculate.R:23]: - [ ]  Add some tests for the other `date_class` option.
- [ ] Add some tests on the return type.
- [ ] Add some test for `round_down = FALSE` (`round_down = TRUE` isn't required here as it's the default.
- [ ] Add a test for start date after end date (i.e. negative age) - Could possibly include an error or warning for this? Maybe also warn for very large (>130) ages?

**Review by Moohan — CHANGES_REQUESTED** [review]
Hey, I've not run any of the code, just checked through it on GitHub. If I get chance to run it I'll add some more comments but probably smarter to wait for you to check and implement any of my suggestions you agree with before I check again.

File: R/age_calculate.R [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+#' # slightly differently to UK law where (in the UK) legally speaking
+#' # leaplings become a year older on the 1st March on non-leap years.
+#' leap1 <- dmy("29-02-2020")
+#' leap2 <- dmy("28-02-2022")
+#' age_calculate(leap1, leap2, date_class = "duration")
+#' age_calculate(leap1, leap2, date_class = "period")
+#' }
+#' @export
+age_calculate <- function(start, end = Sys.time(), units = "years",
+                date_class = c("period", "duration"), round_down = TRUE){
```
[the remark this answers is not in the document]
Comment (Tina815) [reply R/age_calculate.R:47]: I would prefer age_calculate as the name so would rather keep it in this way haha. I think arguments as start and end are clear enough to me so will just leave them.

**Review by Moohan — APPROVED** [review]

**Pull request merged**